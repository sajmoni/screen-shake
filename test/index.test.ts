import { expect, test } from 'vitest'
import createScreenShake, { type ScreenShakeUpdateResult } from '../src'

test('Screen shake', async () => {
  const screenShake = createScreenShake({ seed: 0.7476220180650175 })
  screenShake.add(1)
  const results: ScreenShakeUpdateResult[] = []

  ;[...Array(4)].forEach((_, time) => {
    const result = screenShake.update(time)
    results.push(result)
  })

  expect(results).toMatchSnapshot()
})
