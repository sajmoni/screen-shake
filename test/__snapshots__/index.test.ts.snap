// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Screen shake 1`] = `
[
  {
    "angle": -3.465103646672414,
    "offsetX": -10.395310940017241,
    "offsetY": -10.395310940017241,
  },
  {
    "angle": 4.228331174001096,
    "offsetX": 12.684993522003289,
    "offsetY": 12.684993522003289,
  },
  {
    "angle": -2.1682018827495155,
    "offsetX": -6.504605648248547,
    "offsetY": -6.504605648248547,
  },
  {
    "angle": -0.9398338847766816,
    "offsetX": -2.819501654330045,
    "offsetY": -2.819501654330045,
  },
]
`;
